app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,59 @@
+class PostsController < ApplicationController
+  before_action :set_post, only: [:show, :edit, :update, :destroy]
+  before_action :authenticate_user!, except: [:show,:index]
+
+  def allpost
+    @posts = Post.all
+  end
+
+  def index
+    if current_user
+    @posts = current_user.posts
+  else
+    @posts = Post.all
+  end
+  end
```

Comment:
I see what you're accomplishing with `allpost` and `index`, and your instinct to deviate from RESTful actions is dead-on, but the execution could be a bit more semantic. To me, `allpost` seems to serve the function we expect from a tradition `index`. What if you named `allpost` back to `index`, and gave your non-RESTful name to the action with the conditional? Something like `user_posts`? You could also look into filters if you want to get fancy...